src/reader/decoder.rs

```diff
@@ -904,6 +894,46 @@ impl StreamingDecoder {
         Ok(())
     }
 
+    fn read_application_extension(&mut self) -> Option<Decoded> {
+        // the parser removes sub-block lenghts, so app name and data are concatenated
+        if let Some(&[first, second, ..]) = self.ext.data.strip_prefix(EXT_NAME_NETSCAPE) {
```

Comment:
It seems a tad inconsistent to match the exact subblock length as part of the byte prefix, but _also_ match with an `..` elision. Can we really trust all netscape data application data to be formatted into exactly those subblocks?

Reply:
Fair point, according to Wikipedia, every NETSCAPE extension has just three bytes: https://en.wikipedia.org/wiki/GIF#Animated_GIF.

I rewrote the code to make sure it's also really just three bytes.